stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java

```diff
@@ -0,0 +1,171 @@
+                if (dir != null) {
+                    mkdirs(outdir, dir);
+                }
+
+                extractFile(zin, outdir, name);
+            }
+        } catch (IOException e) {
+            log.debug("I/O Exception when processing files ", e);
+            log.error("I/O Exception when processing files " + e.getMessage());
+        } finally {
+            try {
+                if (zin != null) {
+                    zin.close();
+                }
+            } catch (IOException e) {
+                log.debug("I/O Exception when closing the input stream ", e);
+                log.error("I/O Exception when closing the input stream " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Extract files from the zipInputStream.
+     *
+     * @param in     zipInputStream object
+     * @param outdir output directory file
+     * @param name   name of the file
+     */
+    private static void extractFile(ZipInputStream in, File outdir, String name) {
+        byte[] buffer = new byte[4096];
+        BufferedOutputStream out = null;
+        try {
+            out = new BufferedOutputStream(new FileOutputStream(new File(outdir, name)));
+
+            int count = -1;
+            while ((count = in.read(buffer)) != -1) {
+                out.write(buffer, 0, count);
+            }
+        } catch (FileNotFoundException e) {
+            log.debug("File not found to process " + outdir, e);
+            log.error("File not found to process " + e.getMessage());
+        } catch (IOException e) {
+            log.debug("I/O Exception when closing the input stream " + outdir, e);
+            log.error("I/O Exception when closing the input stream " + e.getMessage());
+        } finally {
+            if (out != null) {
+                try {
+                    out.close();
+                } catch (IOException e) {
+                    log.debug("I/O Exception when closing the input stream ", e);
+                    log.error("I/O Exception when closing the input stream " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Create the directory name.
+     *
+     * @param name name of the directory
+     * @return directory name
+     */
+    private static String dirpart(String name) {
+        int s = name.lastIndexOf(File.separatorChar);
+        return s == -1 ? null : name.substring(0, s);
+    }
+
+    /**
+     * Make directories if they doesn't exists.
+     *
+     * @param outdir destination file
+     * @param path path of the destination directory
+     */
+    private static boolean mkdirs(File outdir, String path) {
+        File d = new File(outdir, path);
+        return !d.exists() && d.mkdirs();
+    }
+
+    @Override
+    public BValue[] execute(Context context) {
+        byte[] content = getBlobArgument(context, SRC_AS_BYTEARRAY_FIELD_INDEX);
+        String destDir = getStringArgument(context, DEST_PATH_FIELD_INDEX);
```

Comment:
destDir can be null.
Do null check in all places where a String is passed as an argument to the function.
Applicable for all the functions.